airbyte-cdk/python/unit_tests/sources/mock_server_tests/test_mock_server_abstract_source.py

```diff
@@ -0,0 +1,418 @@
+            url=f"https://api.airbyte-test.com/v1/{self._resource}",
+            query_params=query_params,
+        )
+
+
+def _create_catalog(names_and_sync_modes: List[tuple[str, SyncMode]]) -> ConfiguredAirbyteCatalog:
+    catalog_builder = CatalogBuilder()
+    for stream_name, sync_mode in names_and_sync_modes:
+        catalog_builder.with_stream(name=stream_name, sync_mode=sync_mode)
+    return catalog_builder.build()
+
+
+def _create_dividers_request() -> RequestBuilder:
+    return RequestBuilder.dividers_endpoint()
+
+
+def _create_legacies_request() -> RequestBuilder:
+    return RequestBuilder.legacies_endpoint()
+
+
+def _create_planets_request() -> RequestBuilder:
+    return RequestBuilder.planets_endpoint()
+
+
+def _create_users_request() -> RequestBuilder:
+    return RequestBuilder.users_endpoint()
+
+
+RESPONSE_TEMPLATE = {
+    "object": "list",
+    "has_more": False,
+    "data": [
+        {
+            "id": "123",
+            "created_at": "2024-01-01T07:04:28.000Z"
+        }
+    ]
+}
+
+USER_TEMPLATE = {
+    "object": "list",
+    "has_more": False,
+    "data": [
+        {
+            "id": "123",
+            "created_at": "2024-01-01T07:04:28.000Z",
+            "first_name": "Paul",
+            "last_name": "Atreides",
+        }
+    ]
+}
+
+PLANET_TEMPLATE = {
+    "object": "list",
+    "has_more": False,
+    "data": [
+        {
+            "id": "456",
+            "created_at": "2024-01-01T07:04:28.000Z",
+            "name": "Giedi Prime",
+        }
+    ]
+}
+
+LEGACY_TEMPLATE = {
+    "object": "list",
+    "has_more": False,
+    "data": [
+        {
+            "id": "l3g4cy",
+            "created_at": "2024-02-01T07:04:28.000Z",
+            "quote": "What do you leave behind?",
+        }
+    ]
+}
+
+DIVIDER_TEMPLATE = {
+    "object": "list",
+    "has_more": False,
+    "data": [
+        {
+            "id": "l3t0",
+            "created_at": "2024-02-01T07:04:28.000Z",
+            "divide_category": "dukes",
+        }
+    ]
+}
+
+
+RESOURCE_TO_TEMPLATE = {
+    "dividers": DIVIDER_TEMPLATE,
+    "legacies": LEGACY_TEMPLATE,
+    "planets": PLANET_TEMPLATE,
+    "users": USER_TEMPLATE,
+}
+
+
+def _create_response() -> HttpResponseBuilder:
+    return create_response_builder(
+        response_template=RESPONSE_TEMPLATE,
+        records_path=FieldPath("data"),
+        # pagination_strategy=StripePaginationStrategy()
+    )
+
+
+def _create_record(resource: str) -> RecordBuilder:
+    return create_record_builder(
+        response_template=RESOURCE_TO_TEMPLATE.get(resource),
+        records_path=FieldPath("data"),
+        record_id_path=FieldPath("id"),
+        record_cursor_path=FieldPath("created_at"),
+    )
+
+
+class FullRefreshStreamTest(TestCase):
+    @HttpMocker()
+    def test_full_refresh_sync(self, http_mocker):
+        start_datetime = _NOW - timedelta(days=14)
+        config = {
+            "start_date": start_datetime.isoformat()[:-13]+"Z",
```

Comment:
This is dangerous since if _NOW have microseconds == 0, this will give a unexpected format. 

```
>>> datetime(2024, 1, 1, 2, 3, 4).isoformat()
'2024-01-01T02:03:04'

>>> datetime(2024, 1, 1, 2, 3, 4, 5).isoformat()
'2024-01-01T02:03:04.000005'
```

We should probably define the format we expect instead